src/main/java/frc/robot/controls/GameControllerDriverConfig.java

```diff
@@ -12,22 +27,68 @@
  */
 public class GameControllerDriverConfig extends BaseDriverConfig {
   private final GameController controller = new GameController(Constants.DRIVER_JOY);
+  private final DriveSub sub;
+
+  public GameControllerDriverConfig(Drivetrain drive, DriveSub sub) {
+      super(drive);
+      this.sub = sub;
+
 
-  public GameControllerDriverConfig(Drivetrain drive) {
-    super(drive);
   }
 
-  @Override
-  public void configureControls() {
+    @Override
+public void configureControls() {
     // TODO 4.1.1: Change to your auto command
-    controller.get(Button.A).onTrue(new DoNothing());
+    controller.get(Button.LB).onTrue(new AutoForwardCommand(drive, -1.0, 3.0));
+    controller.get(Button.RB).onTrue(new AutoForwardCommand(drive, 1.0, 3.0));
+    controller.get(Button.B).onTrue(new AutoSquareCommand(drive));
+    controller.get(Button.X).onTrue(new AutoTurnCommand(drive, 152.17)); //turns 180 degrees (angles for some reason broken)
+
     // TODO 4.1.3: Add Bang-Bang drive command
+    controller.get(Button.START).onTrue(new BangBangSubsystemCommand(sub, 1.0));
+    controller.get(Button.BACK).onTrue(new BangBangRotateCommand(sub, 3.0));
 
     // TODO 4.1.4: Add subsystem Bang-Bangs
+    controller.get(Button.START).onFalse(new BangBangSubsystemCommand(sub, 0));
 
     // TODO 4.2.2: Make robot spin while a button is pressed
+    controller.get(Button.A).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, 1.0), drive));
+    controller.get(Button.Y).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, -1.0), drive));
 
     // TODO 4.3.1: Add more triggers
+
+    // SequentialCommandGroup: Drive forward, then turn 45 degrees
+    controller.get(Button.RIGHT_JOY).onTrue(new SequentialCommandGroup(
+      new RunCommand(() -> drive.arcadeDrive(0, 0), drive),    // stop
+      new AutoForwardCommand(drive, 1.0, 3.0)                  // drive forward
+  ));
+    // ParallelCommandGroup: Drive forward and rotate simultaneously
+    controller.get(Button.LEFT_JOY).onTrue(new ParallelCommandGroup(
+        new AutoForwardCommand(drive, 1.0, 2.0),
+        new BangBangRotateCommand(sub, 1.0)
+    ));
+
+    // ConditionalCommand: Drive forward if LEFT_TRIGGER_BUTTON is pressed, otherwise turn
+new Trigger(controller.LEFT_TRIGGER_BUTTON).onTrue(
+    new ConditionalCommand(
```

Comment:
If your Trigger uses LEFT_TRIGGER_BUTTON, your condition cannot also be LEFT_TRIGGER_BUTTON. You can leave this as new Trigger() or use controller.get() instead. They do the same thing for BooleanSuppliers. Also, since LEFT_TRIGGER_BUTTON is already a BooleanSupplier, which is a type of lambda expression, you do not need to include the ()-> or the .getAsBoolean().

Reply:
I think you still have RIGHT_TRIGGER_BUTTON and a few of the DPad buttons left to use as conditions, or you could change the Trigger to use a DPad button.

Reply:
should work